### ISSUE TITLE
Question: Does it make sense to support outputting arbitrary colors?

### ISSUE THREAD

**Issue by ldionne** [agauniyal/rang#42]
Would it make sense to allow outputting arbitrary colors specified by a hex color code? I'm not sure whether that can even be implemented (i.e. whether terminals even support it), so this is more a question than a feature request at this point.

**Comment by ldionne** [comment]
Thanks for the answer.

Don't consider this a feature request; I just wanted to know if it made sense, and if so bring it to your attention because it's an obvious generalization of being able to output predefined colors only.

**Comment by agauniyal** [comment]
I'll work on a separate function under the same namespace and test it on a few machines in next few days. If everything works out, I'll release a newer version of rang with said feature.

**Comment by agauniyal** [comment]
@ldionne So I've tested a prototype that supports both 256 colors and 24 bit truecolor(rgb mode) but the problem that remains is - there is no way to test if a terminal emulator supports it or not. Some of them do, but we still cannot hardcode them because they managed to support it after certain version which they might or might not expose via env variables.

tldr; it works but there is no way to know if certain terminal supports it or not.
